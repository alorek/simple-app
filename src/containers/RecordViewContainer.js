import React, { Component } from 'react';
import RecordView from '../components/RecordView';
import Button from '../components/Button';

class RecordViewContainer extends Component {
    render() {
        const recordId = this.props.match.params.id;
        const goBack = () => {
            this.props.history.goBack();
        };
        const style = {
            margin: '0 auto',
            maxWidth: '800px'
        };

        return (
            <div className="content-wrapper">
                <div className="record-wrapper" style={ style }>
                    <Button
                        label="Back to records"
                        actionFunc={ goBack }
                        className="btn btn-primary"
                        iconClass="glyphicon glyphicon-arrow-left" />
                    <RecordView recordId={ recordId } />
                </div>
            </div>
        );
    }
}

export default RecordViewContainer;
