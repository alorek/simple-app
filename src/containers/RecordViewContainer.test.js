import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RecordViewContainer from './RecordViewContainer';
import reducer from '../reducers';

const store = createStore(reducer);
const match = {params: {id: '123'}};

test('RecordViewContainer renders correctly', () => {
    const tree = renderer.create(
        <Provider store={ store }>
            <Router>
                <RecordViewContainer match={match}/>
            </Router>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
