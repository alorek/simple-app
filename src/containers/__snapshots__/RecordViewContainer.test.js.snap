// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordViewContainer renders correctly 1`] = `
<div
  className="content-wrapper"
>
  <div
    className="record-wrapper"
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": "800px",
      }
    }
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      <span
        className="glyphicon glyphicon-arrow-left"
        style={
          Object {
            "marginRight": "5px",
          }
        }
      />
      <span>
        Back to records
      </span>
    </button>
    <div>
      <p>
        Record not found
      </p>
    </div>
  </div>
</div>
`;
